### DISCUSSION TITLE
Customize filenames?

### DISCUSSION THREAD

**Discussion by melithine** [Avnsx/fansly#177] [Q&A]
Is there any ability to customize the filenames? It would be particularly helpful for me if i could store the creator username as part of the filename, as that would allow me to more readily manipulate it with the script that I have to upload things to Plex on my TrueNAS system.

A similar program, OF-DL, supports a set of variables that can be used in a template for filenames like so: https://github.com/sim0n00ps/OF-DL/wiki/Custom-Filename-Formats

**Comment by FarmingCrops** [comment]
having a filename like this would be much simpler
2021-06-09 fansly-0zgt80q0vv9p


having a filename like this is unnecessary
2023-02-21_at_11-13_preview_id_484373572094009344_hash_20c42ef7f5189b434330a8042b76dc0f


{
  "DownloadAvatarHeaderPhoto": true,
  "DownloadPaidPosts": true,
  "DownloadPosts": true,
  "DownloadArchived": true,
  "DownloadStreams": true,
  "DownloadStories": true,
  "DownloadHighlights": true,
  "DownloadMessages": true,
  "DownloadPaidMessages": true,
  "DownloadImages": true,
  "DownloadVideos": true,
  "DownloadAudios": true,
  "IncludeExpiredSubscriptions": true,
  "SkipAds": true,
  "DownloadPath": "",
  "PaidPostFileNameFormat": "",
  "PostFileNameFormat": "{mediacreatedAt} {filename}",
  "PaidMessageFileNameFormat": "{mediaCreatedAt} {filename}",
  "MessageFileNameFormat": "{mediaCreatedAt} {filename}",
  "RenameExistingFilesWhenCustomFormatIsSelected": true,
  "Timeout": null,
  "FolderPerPaidPost": false,
  "FolderPerPost": false,
  "FolderPerPaidMessage": false,
  "FolderPerMessage": false,
  "LimitDownloadRate": false,
  "DownloadLimitInMbPerSec": 0,
  "DownloadOnlySpecificDates": false,
  "DownloadDateSelection": "after",
  "CustomDate": null,
  "ShowScrapeSize": true
}
